geonode/base/api/permissions.py

```diff
@@ -227,20 +227,28 @@ class UserHasPerms(DjangoModelPermissions):
         'DELETE': [f'base.{x}' for x in BASIC_MANAGE_PERMISSIONS],
     }
 
+    def __init__(self, perms_dict={}):
+        self.perms_dict = perms_dict
+
+    def __call__(self):
+        return self
+
     def has_permission(self, request, view):
         from geonode.base.models import ResourceBase
 
         queryset = self._queryset(view)
-        perms = self.get_required_permissions(request.method, queryset.model)
+        perms = self.perms_dict.get(request.method, None) or self.get_required_permissions(request.method, queryset.model)
 
         if request.user.is_superuser:
             return True
 
         if view.kwargs.get('pk'):
             # if a single resource is called, we check the perms for that resource
-            res = ResourceBase.objects.filter(pk=view.kwargs.get('pk')).first()
-            if not res:
-                raise NotFound
+            res = get_object_or_404(ResourceBase, pk=view.kwargs.get('pk'))
+            # if the request is for a single resource, we take the specific or the default. If none is defined we keep the original one defined above
+            _specific_perms = self.perms_dict.get(res.get_real_instance().resource_type, self.perms_dict.get('default', {}))
```

Comment:
@mattiagiupponi for clarity, what about renmaing `_specific_perms` to `resource_type_specific_perms`?

Reply:
Renamed